tests/TestRunner/tmp_cluster.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
+
+from local_cluster import LocalCluster
+from argparse import ArgumentParser
+from random import choice
+from pathlib import Path
+import shutil
+import subprocess
+import sys
+
+class TempCluster:
+    def __init__(self, build_dir: str):
+        self.build_dir = Path(build_dir)
+        assert self.build_dir.exists(), "{} does not exist".format(build_dir)
+        assert self.build_dir.is_dir(), "{} is not a directory".format(build_dir)
+        tmp_dir = self.build_dir.joinpath(
+            'tmp',
+            ''.join(choice(LocalCluster.valid_letters_for_secret) for i in range(16)))
+        tmp_dir.mkdir(parents=True)
+        self.cluster = LocalCluster(tmp_dir,
+                                    self.build_dir.joinpath('bin', 'fdbserver'),
+                                    self.build_dir.joinpath('bin', 'fdbmonitor'),
+                                    self.build_dir.joinpath('bin', 'fdbcli'))
+        self.log = self.cluster.log
+        self.etc = self.cluster.etc
+        self.data = self.cluster.data
+        self.tmp_dir = tmp_dir
+
+    def __enter__(self):
+        self.cluster.__enter__()
+        self.cluster.create_database()
+        return self
+
+    def __exit__(self, xc_type, exc_value, traceback):
+        self.cluster.__exit__(xc_type, exc_value, traceback)
+        shutil.rmtree(self.tmp_dir)
+
+
+if __name__ == '__main__':
+    parser = ArgumentParser()
+    parser.add_argument('--build-dir', metavar='BUILD_DIRECTORY', help='FDB build directory', required=True)
+    parser.add_argument('cmd', metavar="COMMAND", help="Command to run", nargs='+')
+    args = parser.parse_args()
+    errcode = 1
+    with TempCluster(args.build_dir) as cluster:
+        print("log-dir: {}".format(cluster.log))
+        print("etc-dir: {}".format(cluster.etc))
+        print("data-dir: {}".format(cluster.data))
+        print("cluster-file: {}".format(cluster.etc.joinpath('fdb.cluster')))
+        cmd_args = []
+        for cmd in args.cmd:
+            if cmd == '@CLUSTER_FILE@':
```

Comment:
Are these replaced by cmake with `configure_file`? It's not clear to me how this file is used, maybe provide an example of calling the script from the command line in the PR description or class comment.

Reply:
I don't think so - conventionally if that were the case this file would end with `.in`. I think this is more like dependency injection. You can write a test binary or script that needs a cluster_file, and this will invoke your test with `@CLUSTER_FILE@` replaced with an actual cluster file.

I think it's probably worth explaining this in the help text for this script.

Reply:
I added a longer description. But basically you could do something like this:

```
./tmp_cluster.py -b ~/build/foundationdb -- ~/build/foundationdb/bin/fdbcli -C @CLUSTER_FILE@ --exec status
```

This would do the following:
1. Create a random cluster configuration in a random directory in the build directory (with one process)
1. Start fdbmonitor
1. Run the passed command and replace certain strings
1. when the command returns, it will kill the server and delete all generated data

This is an easy way to run any test that needs access to a running fdb